service_profiles.md

```diff
@@ -0,0 +1,111 @@
+# Service Profiles
+
+The RFC introduces the notion of service profiles.  A service profile is an
+object that describes characteristics that a service may have.  Throughout
+this RFC we will extensively use the REST Service Profile as an example.  The
+REST Service Profile describes the characteristics of an HTTP service that
+obeys the tenants of [REST](https://en.wikipedia.org/wiki/Representational_state_transfer).
+
+Service profiles are useful because this allows Linkerd to behave differently
+depending on the characteristics of the service to which it is making requests.
+For example, if Linkerd sends an HTTP GET request to a service that has the
+REST profile, the request is guaranteed to be idempotent and therefore is safe
+for Linkerd to retry.  The types of characteristics that could potentially be
+described by a service profile includes:
+
+* success classification
+* retryability/idempotence
+* retry budgets
+* rate limits
+* ACLs
+* latency objectives/timeouts
+* success rate objectives
+* etc.
+
+Each service can be associated with at most one profile.  While we may one day
+wish to explore extending or overriding profiles, this is currently out of
+scope.  It is important to note that even though it is a service that is
+associated with a profile, this can influence the behavior of callers of that
+service.  In many cases the service and its caller may exist in different
+namespaces and/or have different owners.
+
+Service profiles are designed to be built in an incremental and extensible way.
+The initial implementation will contain very few characteristics but will be
+expanded over time.
+
+## Service Profile Definition
+
+While some characteristics such as retry budgets apply to the service as a
+whole, many characteristics vary depending on the request.  For example,
+idempotence may vary depending on the request's HTTP method or timeouts might
+vary depending on the request's HTTP path.
+
+To accommodate this, a service profile contains some top level characteristics
+that apply to the whole service and an ordered list of request policy objects
+which define the characteristics which apply to certain classes of requests.
+
+```
+message ServiceProfile {
+  repeated RequestPolicy request_policies = 1;
+  RetryBudget retry_budget = 2;
```

Comment:
Is `RetryBudget` where backoff strategy would be included? Should that be specced out now, or are we punting on any sort of backoff?

Reply:
Good question.  I had planned on punting on backoff for now.  But yes, this is probably where it would go.